runtime/v2/shim/util.go

```diff
@@ -85,7 +85,7 @@ func Command(ctx context.Context, config *CommandConfig) (*exec.Cmd, error) {
 func BinaryName(runtime string) string {
 	// runtime name should format like $prefix.name.version
 	parts := strings.Split(runtime, ".")
-	if len(parts) < 2 {
+	if len(parts) < 3 {
```

Comment:
This logic doesn’t seem correct.
The right way would be to check filepath.IsAbs in ctr.

Reply:
My reason for making this change here is to my understanding, the value passed to `--runtime` is evaluated in ShimManager's [resolveRuntimePath](https://github.com/containerd/containerd/blob/731518417e18df7b7f28ecf569c45fff79695cb0/runtime/v2/manager.go#L232) method. This method uses `filepath.IsAbs` to check if the runtime provided is an absolute path to an existing file. 

In `resolveRuntimePath`'s fallthrough logic for when the provided value is not an absolute filepath, the `BinaryName` util is used to check for a value like `io.containerd.runc.v1` and return `containerd-shim-runc-v1` in that case. A string like `./containerd-shim-runc-v2` passed to `BinaryName` results in a parts slice `["" "/containerd-shim-runc-v2"]` of length 2 and will miss this error check on the parts length.

I'm not too familiar with all the possible runtime name formats - if a format like `runc.v1` is acceptable then these changes could be updated to add an additional check for if the first element of parts is `""`

Reply:
> if a format like runc.v1 is acceptable

Yes.  Actually `--runtime aws.firecracker` appears in https://github.com/firecracker-microvm/firecracker-containerd/blob/main/docs/getting-started.md#usage